plutus-scb/src/Control/Concurrent/ServiceAvailability.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+module Control.Concurrent.ServiceAvailability where
+
+import           Control.Concurrent (MVar, putMVar, takeMVar)
+
+-- | A semaphore-like construct whereby a service can signal to
+-- another thread that it's finished its startup phase, and is now
+-- available to use.
+class ServiceAvailability m a where
```

Comment:
This works for now but I think at some point we may need a way to manage dependencies with error handling and config changes (hopefully using a library that exists already)

Reply:
Also I think a newtype would be better since there is only one instance of this class and we always pass around the value as an argument anyway

```haskell
newtype Availability = Availability { unAvailability :: MVar () }
starting :: MonadIO m => Availability -> m ()
available :: MonadIO m => Availability -> m ()
```

Reply:
Although I don't have a strong opinion on it

Reply:
🤔 

Yeah, I think you're right. I'll change it. :-)